### DISCUSSION TITLE
Persist query param (?id=asdf) through refresh and new page load

### DISCUSSION THREAD

**Discussion by 65536william** [vercel/next.js#31028] [Help]
I have the page `/[username]/projects` and I don't want individual projects to load in a new page, instead I use the router's query object to identify if a project is selected and fetch the data accordingly. So the link for a project will look like this `/giddysnail4/projects?id=123512`. This all works well.

However, when I refresh the page, or if I access the page by pasting a url in the browser, first the page loads as it should do, with the individual project displaying. However, something about Next.js then forces a redirect to the 'core url' (`/giddysnail4/projects`). Even though the desired detail page appears to load when that page is visited, if I refresh then Next.js drops the query param.

Any ideas on how to prevent the automatic, seemingly forced redirection from a URL with a query param to a non-query param URL?

**Comment by Zw1d** [comment]
I landed on your issue, having a similar one. In my case I caused my own problem: I was setting the url query dynamically depending on checkboxes selected in my form. However, I have not implemented checking the boxes according to url query, so on every reload I was "setting unchecked checkboxes" in the url resulting in empty query string and removal of url parameters, which looked like a redirect.
